NOTES.md

```diff
@@ -0,0 +1,235 @@
+# Librix server API Guide _18/05/2021_
+
+Running locally is set to port 5000.
+
+## Table of Contents
+
+- [Errors](#errors)
+- [Users](#users)
+- [Books](#books)
+- [Matches](#matches)
+
+## Errors
+
+If there is an error in any of the functions we return =>
+
+```
+{
+  "error": {
+    "message": "The error message"
+  }
+}
+```
+
+## Users
+
+All user calls start with /users
+
+### /
+
+**get** => response is an array with the users as objects, this will be deprecated in production of the app
+**post** => expects the following minimal request
+
+```
+{
+  "firstName": "John",
+  "lastName": "Smith",
+  "username": "JohnSmith",
+  "email": "John@Smith.com",
+  "password": "0123abc!@£",
+  "address": {
+	  "city": "Berlin"
+  }
+}
+```
+
+Will recieve the response =>
+
+```
+{
+  "avatar": "/statics/01.png",
+  "points": 0,
+  "booksToOffer": [],
+  "booksToRemember": [],
+  "booksInterestedIn": [],
+  "matches": [],
+  "_id": "60a26618f8d1360733111027",
+  "firstName": "John",
+  "lastName": "Smith",
+  "username": "JohnSmith",
+  "email": "John@Smith.com",
+  "address": {
+    "country": "Germany",
+    "city": "Berlin"
+  },
+  "createdAt": "2021-05-17T12:48:24.125Z",
+  "updatedAt": "2021-05-17T12:48:24.125Z"
+}
+``
+```
+
+### /:id
+
+**get** => response is an the user as an object
+**put** => If the field is made up =>
+
+```
+"This": "is not a field"
+```
+
+Will **not** return an error, just the user as an object not changed in any way. If you try and change a field that is a _mongoose id_, you will recieve an error. Fields that can be changed include =>
+
+```
+{
+  "avatar": "https://cdn.fakercloud.com/avatars/aislinnkelly_128.jpg",
+  "points": 2,
+  "firstName": "Mireya",
+  "lastName": "Armstrong",
+  "username": "Fred32",
+  "email": "Ayden9@hotmail.com",
+  "address": {
+    "country": "Germany",
+    "city": "Berlin"
+  },
+  "createdAt": "2020-05-18T00:00:00.000Z",
+  "updatedAt": "2021-05-18T05:17:48.493Z"
+}
+```
+
+**del** => if deleted successfully you will recieve a response of the user as an object, everything else will throw an error.
+
+### /login
+
+**post** => _to be updated_ => expects the following request =>
+
+```
+{
```

Comment:
we dont have a password?

Reply:
we dont just yet in our app but can be added

Reply:
thanks